version: "3.3"

volumes:
  tosca-db-data:
  tosca-geoserver-data-dir:

services:
  db:
    image: kartoza/postgis:${POSTGIS_VERSION_TAG}
    container_name: tosca-2-db-prod
    volumes:
      - tosca-db-data:/var/lib/postgresql
    ports:
      - ${POSTGRES_PORT}:5432
    environment:
      - POSTGRES_DB=${POSTGRES_DB}
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASS=${POSTGRES_PASS}
      - ALLOW_IP_RANGE=${ALLOW_IP_RANGE}
      - FORCE_SSL=FALSE
    restart: on-failure
    healthcheck:
      test: "PGPASSWORD=${POSTGRES_PASS} pg_isready -h 127.0.0.1 -U ${POSTGRES_USER} -d ${POSTGRES_DB}"

  geoserver:
    build: ./geoserver
    image: tosca-geoserver-image
    container_name: tosca-2-geoserver-prod
    volumes:
      - tosca-geoserver-data-dir:/opt/geoserver/data_dir
      - ./build_data/:/settings
      # sample volume attachment. Please give full path.
      #- /home/tosca/sample_data:/mnt/data
    ports:
      - ${GEOSERVER_PORT}:8080
    restart: on-failure
    env_file:
      - .env
    depends_on:
      - db